Userland/Libraries/LibJS/Console.cpp

```diff
@@ -147,10 +147,47 @@ ThrowCompletionOr<Value> Console::count_reset()
     return js_undefined();
 }
 
-Value Console::assert_()
+// 1.1.1. assert(condition, ...data), https://console.spec.whatwg.org/#assert
+ThrowCompletionOr<Value> Console::assert_()
 {
+    // 1. If condition is true, return.
+    auto condition = vm().argument(0).to_boolean();
+    if (condition)
+        return js_undefined();
+
+    // 2. Let message be a string without any formatting specifiers indicating generically an assertion failure (such as "Assertion failed").
+    auto message = js_string(vm(), "Assertion failed");
+
+    // 3. If data is empty, append message to data.
+    Vector<JS::Value> data;
+    if (vm().argument_count() > 1) {
+        data.ensure_capacity(vm().argument_count() - 1);
+        for (size_t i = 1; i < vm().argument_count(); ++i) {
+            data.append(vm().argument(i));
+        }
+    }
```

Comment:
Maybe this "conversion" should be done before step 2? (Before step 1 seems like a waste)
To make sure that the data is already in the appropriate format.
Right now it looks a bit weird to me since `if (vm().argument_count() > 1) {` is essentially the if of step 3 however it does something completely different.

Just moving the spec text to the if below might already be fine actually.

Reply:
You're right, this is a little nonsensical. I've moved it and added a NOTE comment so it's clear it's not associated with one of the steps.